inline_verifier.go

```diff
@@ -0,0 +1,223 @@
+	return nil
+}
+
+func (v *InlineVerifier) VerifyDuringCutover() (VerificationResult, error) {
+	// TODO: verify everything within the reverify queue.
+	return VerificationResult{}, nil
+}
+
+func (v *InlineVerifier) StartInBackground() error {
+	// not needed?
+	return nil
+}
+
+func (v *InlineVerifier) Wait() {
+	// not needed?
+}
+
+func (v *InlineVerifier) Result() (VerificationResultAndStatus, error) {
+	// not implemented for now
+	return VerificationResultAndStatus{}, nil
+}
+
+func (v *InlineVerifier) CheckFingerprintInline(tx *sql.Tx, targetDb, targetTable string, sourceBatch *RowBatch) ([]uint64, error) {
+	table := sourceBatch.TableSchema()
+
+	fingerprintQuery := table.FingerprintQuery(targetDb, targetTable, sourceBatch.Size())
+	fingerprintStmt, err := v.targetStmtCache.StmtFor(v.TargetDB, fingerprintQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	pks := make([]interface{}, len(sourceBatch.Values()))
+	for i, row := range sourceBatch.Values() {
+		pk, err := row.GetUint64(sourceBatch.PkIndex())
+		if err != nil {
+			return nil, err
+		}
+
+		pks[i] = pk
+	}
+
+	rows, err := tx.Stmt(fingerprintStmt).Query(pks...)
+	if err != nil {
+		return nil, err
+	}
+
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch target data
+	targetFingerprints := make(map[uint64][]byte)                // pk -> fingerprint
+	targetDecompressedData := make(map[uint64]map[string][]byte) // pk -> columnName -> compressedData
+
+	for rows.Next() {
+		rowData, err := ScanByteRow(rows, len(columns))
+		if err != nil {
+			return nil, err
+		}
+
+		pk, err := strconv.ParseUint(string(rowData[0]), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		targetFingerprints[pk] = rowData[1]
+		targetDecompressedData[pk] = make(map[string][]byte)
+
+		for i := 2; i < len(columns); i++ {
+			targetDecompressedData[pk][columns[i]], err = v.DecompressData(table, columns[i], rowData[i])
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Fetch source data
+	sourceFingerprints := sourceBatch.Fingerprints()
+	sourceDecompressedData := make(map[uint64]map[string][]byte)
+
+	for _, rowData := range sourceBatch.Values() {
+		pk, err := rowData.GetUint64(sourceBatch.PkIndex())
+		if err != nil {
+			return nil, err
+		}
+
+		sourceDecompressedData[pk] = make(map[string][]byte)
+		for idx, col := range table.Columns {
+			var compressedData []byte
+			var ok bool
+			if _, ok = table.CompressedColumns[col.Name]; !ok {
+				continue
+			}
+
+			compressedData, ok = rowData[idx].([]byte)
+			if !ok {
+				return nil, fmt.Errorf("cannot convert column %v to []byte", col.Name)
+			}
+
+			sourceDecompressedData[pk][col.Name], err = v.DecompressData(table, col.Name, compressedData)
+		}
+	}
+
+	mismatches := v.CompareHashes(sourceFingerprints, targetFingerprints)
+	if len(mismatches) > 0 {
+		return mismatches, nil
+	}
+
+	mismatches = v.CompareDecompressedData(sourceDecompressedData, targetDecompressedData)
```

Comment:
Actually in a follow up PR I'm guaranteed to call this method every time as I need the exact PKs that mismatched for reverification purposes.

I implemented this because I assume it can be understood that if the sourceDecompressedData and targetDecompressedData are empty, there would be no mismatches. Is this not how you understood this method?

Reply:
I saw that once I looked at the method, but it wasn't obvious at first and I was wondering why it was always called if the above block `continue`d out of the execution path

Reply:
I'm going to leave this code as is now, as I refactor these two method into a standalone method in the next PR where I add the reverifier for the binlog streamer

It'll looke like this: `v.compareHashesAndData(sourceFingerprints, targetFingerprints, sourceDecompressedData, targetDecompressedData)`